### DISCUSSION TITLE
Ability to combine 2 columns into 1

### DISCUSSION THREAD

**Discussion by joannachongshen** [donschia/FetchXmlDetailsList#7] [Ideas]
For example, the Connection table allows connection to either User or Contact table. 

It would be nice to combine the Email address from either **systemuser.internalemailaddress** or **contact.emailaddress1**. 
Out-of-box view:
![image](https://github.com/donschia/FetchXmlDetailsList/assets/115131497/e927cc58-23aa-4d91-a079-c03c6884f5bb)
Combined: 
![image](https://github.com/donschia/FetchXmlDetailsList/assets/115131497/b1c27b6b-3f62-4943-842e-657197ee7e7e)

**Comment by donschia** [comment]
HI Joannachongshen.  I can see how that would be useful, especially for those multi-entity data sources like Connections.  I will have to check into this and let you know.  I suspect it might be a tricky one due to the way the fields are mapped behind the scenes.  But I will look into it.  Thanks for your question regarding FetchXmlDetailsList!

**Comment by donschia** [comment]
Hello again Joannachongshen!
I spent some time on the project today and I believe I have a "combined fields" feature working for this datagrid.   If you get the latest on the [Experimental-Ideas-Aug2023](https://github.com/donschia/FetchXmlDetailsList/tree/Experimental-Ideas-Aug2023) branch you can see it all ready to go.  The sample data shows how it can be used on Connections like you outlined above.   Please take a look when you can and let me know if it works for you.

Essentially you set up the CombinedField in your column layout like the following.  The data.joinValuesFormTheseFields is a list of all field names that are grouped and shown for that column.  If there are values in more than one of the fields, then I show them all (delimited by semicolons -- for now anyhow).  Of course, for mutually exclusive fields (like on the To side of Connections, you won't see more than one show up.

```json
     {
        "key": "CombinedNameField",
        "fieldName": "CombinedNameField",
        "name": "Combined Names",
        "minWidth": 200,
        "data": {
          "joinValuesFromTheseFields": ["contact.fullname","systemuser.fullname"]
        }
      },
      {
        "key": "CombinedEmailField",
        "fieldName": "CombinedEmailField1",
        "name": "Combined Emails",
        "minWidth": 200,
        "data": {
          "joinValuesFromTheseFields": ["contact.emailaddress1","systemuser.internalemailaddress"]
        }
      },
      {
        "key": "CombinedModifiedOnField",
        "fieldName": "CombinedModifiedOnField",
        "name": "Combined ModifiedOn",
        "minWidth": 80,
        "data": {
          "joinValuesFromTheseFields": ["contact.modifiedon","systemuser.modifiedon"],
          "dateFormat": "yyyy-MM-dd hh:mm:ss"
        }
      },
```
The sample FetchXml starts on your Account record and shows any connected Contacts or SystemUsers.
```XML
<fetch top="50">
  <entity name="connection">
    <attribute name="record1id" />
    <attribute name="record2id" />
    <attribute name="record1objecttypecode" />
    <attribute name="record2objecttypecode" />
    <link-entity name="account" from="accountid" to="record1id" alias="account">
      <attribute name="emailaddress1" />
      <attribute name="name" />
      <attribute name="modifiedon" />
      <filter>
        <condition attribute="accountid" operator="eq" value="[RECORDID]" />
      </filter>
    </link-entity>
    <link-entity name="contact" from="contactid" to="record2id" link-type="outer" alias="contact">
      <attribute name="emailaddress1" />
      <attribute name="fullname" />
      <attribute name="modifiedon" />
    </link-entity>
    <link-entity name="systemuser" from="systemuserid" to="record2id" link-type="outer" alias="systemuser">
      <attribute name="internalemailaddress" />
      <attribute name="fullname" />
      <attribute name="modifiedon" />
    </link-entity>
  </entity>
</fetch>
```

I still need to get back to fixing the layout on the test harness, but this is what it renders today after fixing the widths.  Notice I did used this new combined field for Name, Email, and ModifiedOn date just to make sure it worked on the formatted fields.   Seems to be working now.
![image](https://github.com/donschia/FetchXmlDetailsList/assets/19396416/aa593d8c-ee5e-4b79-bf8d-9f94f4c409fb)

As mentioned earlier, the changes and the pre-built solution zip files are now in the latest branch so you can easily try it out.  Or just grab the solution here and let me know:
https://github.com/donschia/FetchXmlDetailsList/tree/Experimental-Ideas-Aug2023/solution/bin/Release

Thanks again for your suggestion!
Don

**Reply by joannachongshen** [reply]
This is great, will definitely try it out. Thank you!
